mocks/agent_mock.go

```diff
@@ -0,0 +1,162 @@
+/*
```

Comment:
This file doesn't need to be committed since we're already generating it in the Makefile?

Reply:
removed from source control

Reply:
The generated proto go file is commited as well, let's commit this one as well.

Two suggestions:
1. we need a verify srcipt to make sure the generated code is up-to-date, similar to this [file](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-codegen.sh) in k/k. We don't need to do it in this PR, but if you agree, let's create an issue to track it.
2. let's move this package next to the real code, i.e., proto/agent/mock/agent_mock.go.

Reply:
Moved to proto/agent/mock/agent_mock.go